game/src/plugins/skills/mining/src/mining.plugin.kts

```diff
@@ -50,23 +60,33 @@ class MiningAction(val player: Player, val p: Position, val ore: Ore) : Distance
         }
     }
 
-    fun findPickaxe(): Pickaxe? { // Find the best pick the player has
+    private fun findPickaxe(): Pickaxe? {
         for (id in getPickaxes()) {
+            val pick = lookupPickaxe(id) //Will not return null
+            if (pick!!.level > mob.skillSet.getSkill(Skill.MINING).currentLevel) {
+                continue;
+            }
             if (mob.equipment.get(EquipmentConstants.WEAPON).id == id) {
-                return lookupPickaxe(id);
+                return pick;
             } else if (mob.inventory.contains(id)) {
-                return lookupPickaxe(id);
+                return pick;
             }
         }
         return null;
     }
 
-    fun startMine(pickaxe: Pickaxe) {
+    private fun startMine(pickaxe: Pickaxe) {
         started = true
         mob.sendMessage("You swing your pick at the rock.")
         mob.playAnimation(pickaxe.animation)
         counter = pickaxe.pulses
     }
+
+    private fun miningSuccessful(oreChance: Double, oreChanceOffset: Boolean, playerLevel: Int): Boolean {
+        val percent = (oreChance * playerLevel + (if (oreChanceOffset) 1 else 0 )) * 100;
```

Comment:
Can you move the if/else to a separate value please

Reply:
Will do